### DISCUSSION TITLE
new input/output scalars?

### DISCUSSION THREAD

**Discussion by Zerebokep** [dotansimha/graphql-code-generator#9494] [Q&A]
The new scalars break my project, and I'm currently not sure if this makes sense for me at all. Is there an easy way to get the v3 scalars back in v4?

**Comment by MarcPorciuncula** [comment]
I had a similar issue, specifically with the `ID` type being either `string` or `number` in some cases. I fixed this by using the `scalars` option and explicitly setting the type of `ID` to `string` only.

```yaml
# codegen.yml
config:
  scalars:
    ID: string
```

I'm using Apollo Server + Apollo Client and as far as I can tell they uniformly treat IDs as `string`, so this is safe for me. I'd assume this change was made to fix compatibility with some other framework so you'll want to double check that this is correct for yours.

**Reply by Zerebokep** [reply]
doesn't seem to work, I've opened an issue: https://github.com/dotansimha/graphql-code-generator/issues/9547

**Comment by sgri** [comment]
I have this issue, too.
